core/auth/src/main/java/software/amazon/awssdk/auth/credentials/AwsCredentialsProviderChain.java

```diff
@@ -91,14 +91,19 @@ public AwsCredentials resolveCredentials() {
             try {
                 AwsCredentials credentials = provider.resolveCredentials();
 
-                log.debug("Loading credentials from {}", provider.toString());
+                if (log.isDebugEnabled()) {
+                    log.debug("Loading credentials from {}", provider);
+                }
```

Comment:
The Logger in the Utils class automatically checks if debug is enabled (uses a lambda for its parameter). We could either switch to that, or probably just remove the `isDebugEnabled` check here. Since we're no longer calling `toString`, there's very little performance gain to the check.

Reply:
We are still calling `toString` here though. Yeah, will switch to the util Logger here.

Reply:
Where are we calling `toString` here?

Reply:
`provider.toString()` will be called when it's being logged, isn't it?

Reply:
I think this interface takes an Object and delays calling toString until after the isDebugEnabled check.